src/blueprint/azext_blueprint/custom.py

```diff
@@ -285,38 +383,45 @@ def update_blueprint_resource_group(client,
         resource_group['depends_on'] = _process_depends_on_for_update(depends_on)
     if tags is not None:
         resource_group['tags'] = tags
-
-    rgs = client.create_or_update(resource_scope=resource_scope,
-                                  blueprint_name=blueprint_name,
-                                  blueprint=body).resource_groups
+    args.setdefault('resource_groups', {})[artifact_name] = resource_group

```

Comment:
It will remove the resource groups in other artifacts. You should pass the whole resource group. But why not use sub commands?